app/models/listing.py

```diff
@@ -65,34 +65,50 @@ def category_name(self):
     @staticmethod
     def search(**kwargs):
         """ Returns all listings matching the criteria """
-        filter_list = []
```

Comment:
The filter list applies each of the filters multiple times over it seems. what would normally be a query that returns 102 results returns only 7